### ISSUE TITLE
Injectable_generator appears not to be working with non-nullable @factoryParams (GetIt v7.2.0)

### ISSUE THREAD

**Issue by gabrielaraujoz** [Milad-Akarie/injectable#267]
HI there!

I'm facing something weird here with injectable: after I have adapted some classes to the new GetIt version, which accepts non-nullable factoryParams, I'm getting this error from injectable_generator:

```
[SEVERE] injectable_generator:injectable_builder on lib/main/di/main_module.dart:

Factory params must be nullable
package:my_app/main/di/main_module.dart:13:50
   ╷
13 │   MainPage getMainPage(@factoryParam Destination childDestination) {
   │                                                  ^^^^^^^^^^^^^^^^
```
and also
```
[SEVERE] injectable_generator:injectable_builder on lib/main/presentation/main_page_view_model.dart (cached):

Factory params must be nullable
package:my_app/main/presentation/main_page_view_model.dart:24:31
   ╷
24 │     @factoryParam Destination childDestination,
   │                               ^^^^^^^^^^^^^^^^
```

My implementation:

```
@module
abstract class MainModule {
  MainPage getMainPage(@factoryParam Destination childDestination) {
    return MainPage(
      getIt<MainPageViewModel>(param1: childDestination), ...
```

```
@injectable
class MainPageViewModel extends _MainPageViewModelBase with _$MainPageViewModel {
  MainPageViewModel(
    String notification,
    @factoryParam Destination childDestination, ...
```

Am I doing anything wrong or is this really something we should take action to correct?

Thanks!

**Comment by gabrielaraujoz** [comment]
@Milad-Akarie I just added a PR that should fix this, could you please tell me WDYT once you have a chance to look into it?

**Comment by Milad-Akarie** [comment]
@gabrielaraujoz the PR is included in version 1.5.3 
Thank you for  your effort.
